### PR TITLE
Add lsd

### PR DESCRIPTION
**Pull request by FelixMoeller3**
Add lsd as an extra benchmark and to get line detection working

### CODE REVIEW

File: gluefactory/models/extractors/jpldd/joint_point_line_extractor.py [hunk 1]
```diff
@@ -237,12 +249,23 @@ def _forward(self, data):
...
-        if not self.training:
-            line_segments = None  # as endpoints
-            output["line_segments"] = line_segments
+        if self.conf.line_detection.do and not self.training:
+            if self.conf.timeit:
+                start_lines = time.time()
+            lines = []
+            np_img = (data['image'].cpu().numpy()[:, 0] * 255).astype(np.uint8)
+            np_df = output["line_distancefield"].cpu().numpy()
+            np_ll = output["line_anglefield"].cpu().numpy()
```
Comment (RaphaelKreft) [inline gluefactory/models/extractors/jpldd/joint_point_line_extractor.py]: I think np_al is a better name

File: gluefactory/models/extractors/jpldd/joint_point_line_extractor.py [hunk 2]
```diff
@@ -237,12 +249,23 @@ def _forward(self, data):
...
+            for img, df, ll in zip(np_img, np_df, np_ll):
+                line = detect_afm_lines(
```
Comment (RaphaelKreft) [inline gluefactory/models/extractors/jpldd/joint_point_line_extractor.py]: I think img_lines is a better name to indicate its multiple lines from one image

File: gluefactory/models/extractors/jpldd/line_detection_lsd.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
+from gluefactory.geometry.line_utils import merge_lines, filter_outlier_lines
+from gluefactory.utils.deeplsd_utils import preprocess_angle
+from pytlsd import lsd
+import numpy as np
+
+
+def detect_afm_lines(img: np.array, df: np.array, line_level: np.array, filtering='normal',
```
Comment (RaphaelKreft) [inline gluefactory/models/extractors/jpldd/line_detection_lsd.py:7]: Is this taken off the shelf from lsd? There are some constants I wonder where they are coming from. (ex gradnorm = np.maximum(5 - df, 0).astype(np.float64))

Comment (FelixMoeller3) [reply gluefactory/models/extractors/jpldd/line_detection_lsd.py:7]: This is taken from deeplsd_inference, yeah. It's practically the same function

**Pull request merged**